roles/cockpit-ui/tasks/main.yml

```diff
@@ -41,37 +41,45 @@
   run_once: true
   when: check_docker_registry_exists.rc == 0
 
-- name: Retrieve docker-registry route
+- name: Retrieve docker-registry json
   oc_route:
     name: docker-registry
     namespace: default
     kubeconfig: "{{ openshift_hosted_kubeconfig }}"
-    key_path:
-      jsonpath: "{.spec.host}"
     state: list
-  register: docker_registry_route
+  register: docker_registry_route_json
   changed_when: false
   when: check_docker_registry_exists.rc == 0
 
-- name: Retrieve cockpit kube url
+- name: Transform docker-registry route json to dict
+  set_fact:
+    docker_registry_route: "{{ docker_registry_route_json|from_json }}"
+  changed_when: false
+  when: docker_registry_route_json is defined
+
+- name: Retrieve cockpit kube json
   oc_route:
     name: registry-console
     namespace: default
     kubeconfig: "{{ openshift_hosted_kubeconfig }}"
-    key_path:
-      jsonpath: "https://{.spec.host}"
     state: list
-  register: registry_console_cockpit_kube_url
+  register: registry_console_cockpit_kube_json
   changed_when: false
   when: check_docker_registry_exists.rc == 0
 
+- name: Transform cockpit kube json to dict
+  set_fact:
+    registry_console_cockpit_kube: "{{ registry_console_cockpit_kube_json|from_json }}"
```

Comment:
I'm not a huge fan of using set_fact, I think it would be better to either just move the filter invocation to where the variable is used, or to set the variable on the task using `vars`